runtime/near-vm-logic/src/tests/bls12_381_verify.rs

```diff
@@ -0,0 +1,476 @@
+            .expect(hex_convert_error_msg);
+    let pubkeys: Vec<Vec<u8>> = vec![
+        <Vec<u8>>::from_hex("a491d1b0ecd9bb917989f0e74f0dea0422eac4a873e5e2644f368dffb9a6e20fd6e10c1b77654d067c0618f6e5a7f79a").expect(hex_convert_error_msg),
+    ];
+
+    let pubkeys_raw: Vec<u8> = pubkeys.concat();
+
+    check_bls12_381_verify(
+        signature.len() as u64,
+        &signature,
+        message.len() as u64,
+        &message,
+        pubkeys_raw.len() as u64,
+        &pubkeys_raw,
+        Ok(1),
+        map! {
+            ExtCosts::read_memory_byte: 176,
+            ExtCosts::bls12381_verify_base: 1,
+            ExtCosts::read_memory_base: 3,
+            ExtCosts::bls12381_verify_byte: 32,
+            ExtCosts::bls12381_verify_elements: 1,
+        },
+    );
+
+    let signature: Vec<u8> = <Vec<u8>>::from_hex("9712c3edd73a209c742b8250759db12549b3eaf43b5ca61376d9f30e2747dbcf842d8b2ac0901d2a093713e20284a7670fcf6954e9ab93de991bb9b313e664785a075fc285806fa5224c82bde146561b446ccfc706a64b8579513cfc4ff1d930").expect(hex_convert_error_msg);
+    let message: Vec<u8> =
+        <Vec<u8>>::from_hex("abababababababababababababababababababababababababababababababab")
+            .expect(hex_convert_error_msg);
+    let pubkeys: Vec<Vec<u8>> = vec![
+        <Vec<u8>>::from_hex("a491d1b0ecd9bb917989f0e74f0dea0422eac4a873e5e2644f368dffb9a6e20fd6e10c1b77654d067c0618f6e5a7f79a").expect(hex_convert_error_msg),
+        <Vec<u8>>::from_hex("b301803f8b5ac4a1133581fc676dfedc60d891dd5fa99028805e5ea5b08d3491af75d0707adab3b70c6a6a580217bf81").expect(hex_convert_error_msg),
+        <Vec<u8>>::from_hex("b53d21a4cfd562c469cc81514d4ce5a6b577d8403d32a394dc265dd190b47fa9f829fdd7963afdf972e5e77854051f6f").expect(hex_convert_error_msg),
+    ];
+
+    let pubkeys_raw: Vec<u8> = pubkeys.concat();
+
+    check_bls12_381_verify(
+        signature.len() as u64,
+        &signature,
+        message.len() as u64,
+        &message,
+        pubkeys_raw.len() as u64,
+        &pubkeys_raw,
+        Ok(1),
+        map! {
+            ExtCosts::read_memory_byte: 272,
+            ExtCosts::bls12381_verify_base: 1,
+            ExtCosts::read_memory_base: 3,
+            ExtCosts::bls12381_verify_byte: 32,
+            ExtCosts::bls12381_verify_elements: 3,
+        },
+    );
+}
+
+#[test]
+fn test_bls12_381_verify_aggregate_valid() {
+    let hex_convert_error_msg = "Error during converting hex string to bytes";
+
+    let signature: Vec<u8> = <Vec<u8>>::from_hex("9712c3edd73a209c742b8250759db12549b3eaf43b5ca61376d9f30e2747dbcf842d8b2ac0901d2a093713e20284a7670fcf6954e9ab93de991bb9b313e664785a075fc285806fa5224c82bde146561b446ccfc706a64b8579513cfc4ff1d930").expect(hex_convert_error_msg);
+    let message: Vec<u8> =
+        <Vec<u8>>::from_hex("abababababababababababababababababababababababababababababababab")
+            .expect(hex_convert_error_msg);
+    let pubkeys_raw: Vec<Vec<u8>> = vec![
```

Comment:
[nit]
aren't types inferred by the compiler?

Reply:
fixed: https://github.com/near/nearcore/pull/8184/commits/aded51abb45ec32f1bdc152a474921f6a84063bf